Mathlib/Algebra/Field/Defs.lean

```diff
@@ -47,93 +45,160 @@ a `GroupWithZero` lemma instead.
 field, division ring, skew field, skew-field, skewfield
 -/
 
+-- `NeZero` should not be needed in the basic algebraic hierarchy.
+-- assert_not_exists NeZero
```

Comment:
@semorrison, how much do you care about this `assert_not_exists`? #11202 (+ previous PRs) is an attempt at keeping it, but that PR not enough and I'm finding it hard to avoid ditching it.

Reply:
Nevermind, I managed to fix it